click_option_group/_decorators.py

```diff
@@ -15,6 +15,11 @@
     raise_mixing_decorators_error,
 )
 
+T = TypeVar('T')
+F = TypeVar('F', bound=abc.Callable)
+
+Decorator = abc.Callable[[F], F]
```

Comment:
Could you use `typing.Callable` instead of `abc.Callable` for compatibility with Python 3.7-3.8? [PEP 585](https://peps.python.org/pep-0585/) was implemented in Python 3.9.

Reply:
Done.

I also made a correction to my `help_option` test.